File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
// sol 1 - GPT4

#include <vector>

class Solution {
public:
    int longestSubarray(std::vector<int>& nums) {
        int left = 0, zero_count = 0, max_length = 0;

        for (int right = 0; right < nums.size(); right++) {
            // If the current element is 0, increment the zero_count
            if (nums[right] == 0) {
                zero_count++;
            }

            // If there are more than one 0's, shrink the window
            while (zero_count > 1) {
                if (nums[left] == 0) {
                    zero_count--;
                }
                left++;
            }

            // Update the maximum length of consecutive 1's (subtracting one for deletion)
            max_length = std::max(max_length, right - left);
        }

        return max_length;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
# Sol1 - GPT4

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        zero_count = 0
        max_length = 0

        for right in range(len(nums)):
            # If the current element is 0, increment the zero_count
            if nums[right] == 0:
                zero_count += 1
            
            # If there are more than one 0's, shrink the window
            while zero_count > 1:
                if nums[left] == 0:
                    zero_count -= 1
                left += 1

            # Update the maximum length of consecutive 1's (subtracting one for deletion)
            max_length = max(max_length, right - left)

        return max_length
